### DISCUSSION TITLE
Making CellProfiler more beginner friendly

### DISCUSSION THREAD

**Discussion by bethac07** [CellProfiler/CellProfiler#4829] [Ideas]
Below are some ideas for friendliness improvements, one per comment so can be individually discussed. Anything we decide to do, we can convert to an issue!

**Comment by bethac07** [comment]
Rather than having our example pipelines downloadable from our website, what if we also packaged just the pipelines and existing screenshots into a pretty welcome-screen-acessible-interface the application (text files, would be negligible size increase) and make that an option you can grab at the welcome screen, and then if the user wants to download the images too, have a button for that. In a way, it would be nice for us, because we could see which examples are most used (by tracking downloads), and an easier "onboarding" for users.

**Comment by bethac07** [comment]
A "search the forum" help button could be cool.

**Reply by emiglietta** [reply]
is it possible to integrate it with the error messages specifically? like, having an additional button on the error message that automatically takes you to similar posts in the forum? That is a good way of also having a main hub for answering common issues without repeating so much.

**Comment by bethac07** [comment]
In the same way we have simple/advanced mode for IDPrimary and Watershed, is it [birds or national parks](https://xkcd.com/1425/) to have a q&a wizard for the input modules? I kind of feel like it's parks, and might be one of the best things we could do for making opening CellProfiler less scary for beginners. Existing modules would stay for experts or for tweaking, aka an advanced mode.

**Comment by sugan89** [comment]
One of my friends who used CP mentioned that it would be nice to have `before` and 'after' images (like we have in the photo editing tools) to compare if the previous/current parameters are better for segmentation.

**Comment by sugan89** [comment]
A `single click` option that enables/disables the display of the output of all modules. This will be especially helpful when there are a lot of modules in the pipeline.

**Reply by braymp** [reply]
Does the _Windows > Show/Hide All Windows On Run_ menu item satisfy this item?

**Reply by sugan89** [reply]
Yes! Sorry for my ignorance.

**Reply by gnodar01** [reply]
Update to this one: not yet pulled, but when it is, you can shift-select the subset of modules you want to disable, right click, and disable that subset only.

**Comment by bethac07** [comment]
Giving individual module instantiations different display names - requested externally and internally many, many times (#820 , #2370), etc. If we are going towards a more structured internal pipeline representation, it should become more feasible to give modules a display name that doesn't conflict with its identity

**Reply by bethac07** [reply]
Another proposed version is with color (#2133)

**Comment by bethac07** [comment]
Module groups (discussed internally today, also #4513, related to above issue re-renaming, and others) - some visual-or-otherwise way of saying "these X modules go together for logical function Y, let me turn them on and off in a suite and/or just see that they go together". As above, if going to a more machine-y internal pipeline representation, should make it easier, because group is just another key in the module description dictionary.

As with above one, there are and have always been two issues - the tracking, and then also the UI (this has a third, about the mechanic for turning groups of modules on and off, but again, if it's just a key in a module's "header", unless I radically misunderstand the internal pipeline logic (which to be fair, I never rule out), 'set module 2 to enabled=False' isn't conceptually that much harder than 'set all modules where group=C to enabled=False'). I could imagine for this one that especially for beginners, this concept could be confusing, so possibly could require going into preferences to enable, and then basically do what we do for the module tracing, but to the right of the module names rather than the left. I think to simplify UI, it would also be reasonable to say we only group consecutive sets of modules (ie you can group 2-4 or 2-5 but not 2,4,5; possibly we can help there with display naming and/or coloring as discussed above)

**Comment by bethac07** [comment]
Make it optional/or just new default to have all metadata columns at the beginning of the spreadsheet, not mixed in with data columns

**Comment by emiglietta** [comment]
To make it easier to load and work with multichannel images, maybe have a pop-up suggestion to use the ColorToGray module immediately after? I know the information is in the help button, but I've seen confusion there more than once. Also maybe think about renaming the ColorToGray module to SplitColorImage or sth similar?

**Comment by emiglietta** [comment]
Would it be possible to add a new Module Category that was "Suggested modules" based on the pipeline build so far? Or a button that, based on the position in the pipeline and the modules used before, would suggest a possible next module? I don't think it should be nothing fancy (not a AI-based prediction of any sort), just that:
- if you uploaded color images, maybe you want to split them
- after input, maybe you want to preprocess
- after preprocessing, you likely want to identify primary, and then secondary
- if you input 3D images, you cant to threshold and then watershed
- and so on
